File: benchmarks/pandas/bench_dataframe_cummin.py
import pandas as pd, time, json
N = 100_000
cols = 4
data = {f"col{c}": [(i % 100) * 1.0 for i in range(N)] for c in range(cols)}
df = pd.DataFrame(data)
WARMUP = 3
ITERS = 20
for _ in range(WARMUP):
    df.cummin()
t0 = time.perf_counter()
for _ in range(ITERS):
    df.cummin()
total = (time.perf_counter() - t0) * 1000
print(json.dumps({"function": "dataframe_cummin", "mean_ms": total / ITERS, "iterations": ITERS, "total_ms": total}))

File: benchmarks/tsb/bench_dataframe_cummin.ts
import { dataFrameCummin } from "tsb";
import { DataFrame } from "tsb";
const N = 100_000;
const cols = 4;
const data: Record<string, number[]> = {};
for (let c = 0; c < cols; c++) {
  data[`col${c}`] = Array.from({ length: N }, (_, i) => (i % 100) * 1.0);
}
const df = new DataFrame(data);
const WARMUP = 3;
const ITERS = 20;
for (let i = 0; i < WARMUP; i++) dataFrameCummin(df);
const t0 = performance.now();
for (let i = 0; i < ITERS; i++) dataFrameCummin(df);
const total = performance.now() - t0;
console.log(JSON.stringify({ function: "dataframe_cummin", mean_ms: total / ITERS, iterations: ITERS, total_ms: total }));
